### PR TITLE
Persist charts to cosmos db

### PR DESCRIPTION
**Pull request by sprucely**
I added elastic logging because cosmos db seems a bit finicky, or at least the local emulator does. It gave a 500 Internal Error about half of the times it was requested to create the db/container.

I added a `UseGateway` option, because the VPN firewall doesn't seem to allow `Direct` connections from dev machines to the Azure Cosmos instance, and debugging against the local emulator does not work with `Gateway` connections.

### CODE REVIEW

File: Finjector.Core/Services/CosmosDbService.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        Log.Information("Adding or updating chart {ChartId}", chart.Id);
+        var client = await _cosmosClient.Value;
+        var container = client.GetContainer(_cosmosOptions.DatabaseName, ChartContainerName);
+        var partitionKey = new PartitionKey(chart.IamId);
+        var response = await container.UpsertItemAsync(chart, partitionKey);
+    }
+
+    public async Task DeleteChart(string id, string iamId)
+    {
+        Log.Information("Deleting chart {ChartId}", id);
```
Comment (jSylvestre) [inline Finjector.Core/Services/CosmosDbService.cs:84]: When Logging, include the iamId

**Review by jSylvestre — CHANGES_REQUESTED** [review]
Runs fin locally, but add IAM to logging.
And the pipeline build failed.

**Review by jSylvestre — APPROVED** [review]

**Review by jSylvestre — APPROVED** [review]
Didn't test, but looks sound.

File: Finjector.Web/Handlers/IamIdClaimFallbackTransformer.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                return principal;
+            }
+
+            // if we don't have an IAM ID, try to get it from kerberos and save it in the claims
+            var kerbId = principal.FindFirstValue(ClaimTypes.NameIdentifier);
+
+            if (string.IsNullOrWhiteSpace(kerbId))
+            {
+                return principal;
```
Comment (jSylvestre) [inline Finjector.Web/Handlers/IamIdClaimFallbackTransformer.cs:33]: Throw an exception if the kerb is missing?

Comment (sprucely) [reply Finjector.Web/Handlers/IamIdClaimFallbackTransformer.cs:33]: The controller already returns a `NotAuthorized` if it fails to find an iamid claim. I figured that was good enough.

**Review by jSylvestre — APPROVED** [review]
Didn't test, but looks good.

**Pull request merged**